pyanaconda/ui/gui/spokes/software_selection.py

```diff
@@ -129,6 +132,34 @@ def __init__(self, *args, **kwargs):
         self._fake_radio = Gtk.RadioButton(group=None)
         self._fake_radio.set_active(True)
 
+        # Display a group of options for selecting desired properties of a kernel.
+        self.kernel_box = self.builder.get_object("kernelBox")
+        self.check_button_redhat_kernel = self.builder.get_object("redhatKernelCheckButton")
+        self.check_button_64k = self.builder.get_object("64kCheckButton")
+        self._init_kernel_properties()
+
+    def _init_kernel_properties(self):
+        """Display options for selecting kernel properties."""
+        # Only showing parts of kernel box relevant for current system.
+        show_64k = is_arm() and conf.ui.show_arm_64k_kernel()
+        show_redhat = conf.ui.show_redhat_kernel()
```

Comment:
So I checked how difficult it would be verify the availability of the packages, so you don't have to blindly rely on the configuration settings, and maybe it is worth it. Upstream has the following function that you could copy to RHEL 9 and, during the backport to upstream (if it is planned), this function can be easily be exposed via DBus API.

https://github.com/rhinstaller/anaconda/blob/ea679e9aa9d5bee788c918e4e61ba208bb23b1d6/pyanaconda/modules/payloads/payload/dnf/dnf_manager.py#L496-L507

So I am suggesting to drop the configuration options and check the availability of packages to improve the user experience. What do you think, @jkonecny12?

Reply:
I like this solution. I'm always for not introducing a new API :).